src/NServiceBus.Core/Transports/Sending.cs

```diff
@@ -1,53 +1,60 @@
+#pragma warning disable 1591
 namespace NServiceBus
 {
-    using System;
-    using System.Threading;
-    using System.Threading.Tasks;
-    using Features;
+    using ObjectBuilder;
+    using Routing;
+    using Settings;
     using Transport;
 
-    class Sending : Feature
+    public class TransportComponent
     {
-        public Sending()
+        public TransportComponent(TransportInfrastructure transportInfrastructure)
         {
-            EnableByDefault();
+            TransportInfrastructure = transportInfrastructure;
         }
 
-        protected internal override void Setup(FeatureConfigurationContext context)
-        {
-            var transport = context.Settings.Get<OutboundTransport>();
-            var lazySendingConfigResult = new Lazy<TransportSendInfrastructure>(() => transport.Configure(context.Settings), LazyThreadSafetyMode.ExecutionAndPublication);
-            context.Container.ConfigureComponent(c =>
-            {
-                var dispatcher = lazySendingConfigResult.Value.DispatcherFactory();
-                return dispatcher;
-            }, DependencyLifecycle.SingleInstance);
+        public IDispatchMessages Dispatcher { get; set; }
```

Comment:
any idea how to achieve this? At some point you need to access the "real" dispatcher and there is no transitioning phase we can use without chaining factories again (then we can actually just stay with the DI container?).